examples/Progressor/Client/ResponseProgress/ResponseProgress.cs

```diff
@@ -0,0 +1,65 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Grpc.Core;
+
+namespace Client.ResponseProgress
+{
+    public class ResponseProgress<TMessage, TResult, TProgress> : Progress<TProgress> where TMessage : IProgressMessage<TResult, TProgress>
```

Comment:
`TMessage` is required because `IAsyncStreamReader<T>` is not covariant.

Reply:
https://github.com/grpc/grpc/pull/20549

Reply:
There's a new Grpc.Core.Api release, I think we can update to that and make the change?

Reply:
Still waiting on 2.25.0-pre1 of Grpc.AspNetCore here